internal/testing/pruning_test.go

```diff
@@ -274,7 +278,40 @@ func TestSectorPruning(t *testing.T) {
 	// assert amount of prunable data
 	if n, err := b.PrunableData(context.Background()); err != nil {
 		t.Fatal(err)
-	} else if n != int64(5)*int64(rs.TotalShards)*rhpv2.SectorSize {
+	} else if n != int64(math.Ceil(float64(numObjects)/2))*int64(rs.TotalShards)*rhpv2.SectorSize {
 		t.Fatal("unexpected prunable data", n)
 	}
+
+	// prune all contracts
+	for _, c := range contracts {
+		if _, err := w.RHPPruneContract(context.Background(), c.ID); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// assert spending records were updated and prunable data is 0
+	if err = Retry(10, testBusFlushInterval, func() error {
+		if n, err := b.PrunableData(context.Background()); err != nil {
+			t.Fatal(err)
+		} else if n != 0 {
+			return fmt.Errorf("unexpected prunable data: %d", n)
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	// assert spending was updated
+	for _, c := range contracts {
+		c, err := b.Contract(context.Background(), c.ID)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if c.Spending.SectorRoots.IsZero() {
+			t.Fatal("spending record not updated")
+		}
+		if c.Spending.Deletions.IsZero() {
+			t.Fatal("spending record not updated")
+		}
+	}
```

Comment:
I think we should try to download all objects and make sure only the deleted ones fail. That's the least we can do to verify that the pruning didn't delete the wrong data from the contract.
Ideally we'd check the actual roots of the contracts before and after the pruning.

Reply:
going to make this change in the next PR that adds the cache